src/main/java/omero/cmd/graphs/DuplicateI.java

```diff
@@ -660,6 +827,171 @@ private void linkToNewDuplicates() throws GraphException {
+            duplicateTemplatePrefix = managedRepository.createTemplatePath(idMap, ec, sf, session, sql);
+        } catch (ServerError se) {
+            final String message = "failed to create managed repository directory for duplicate fileset";
+            LOGGER.warn(message, se);
+            throw new GraphException(message + ": " + se);
+        }
+        duplicateFileset.setTemplatePrefix(duplicateTemplatePrefix.toString() + FsFile.separatorChar);
+        managedRepositoryRewrites.put(originalTemplatePrefix, duplicateTemplatePrefix);
+        LOGGER.debug("will duplicate Fileset:{} as Fileset:{} into {}",
+                originalFileset.getId(), duplicateFileset.getId(), duplicateTemplatePrefix);
+    }
+
+    /**
+     * Note the references to underlying files that need fixing in duplicates.
+     */
+    private void noteWrongPaths() throws GraphException {
+        final SqlAction sql = helper.getSql();
+        final String repoUuid = managedRepository.getRepoUuid();
+        for (final Map.Entry<IObject, IObject> originalDuplicate : originalsToDuplicates.entrySet()) {
+            final IObject original = originalDuplicate.getKey();
+            final IObject duplicate = originalDuplicate.getValue();
+            if (original instanceof Fileset) {
+                duplicateFileset((Fileset) original, (Fileset) duplicate);
+            } else if (original instanceof OriginalFile) {
+                final OriginalFile file = (OriginalFile) original;
+                final Path path = getPath(file);
+                if (path != null && isPathManaged.test(path)) {
+                    wrongPathDuplicateFiles.add((OriginalFile) duplicate);
+                }
+            } else if (original instanceof Pixels) {
+                final List<String> pixelInfo = sql.getPixelsNamePathRepo(original.getId());
+                if (pixelInfo != null) {
+                    unmappedPropertiesPixels.put(duplicate.getId(), pixelInfo);
+                    if (repoUuid.equals(pixelInfo.get(2))) {
+                        wrongPathDuplicatePixels.add(duplicate.getId());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Fix the path set for duplicated files in the managed repository such that they reference their new fileset.
+     * @throws GraphException if the paths for the files could not be created
+     */
+    private void fixPathDuplicateFiles() throws GraphException {
+        final EventContext ec = helper.getEventContext();
+        final ServiceFactory sf = helper.getServiceFactory();
+        final Session session = helper.getSession();
+        final SqlAction sql = helper.getSql();
+        for (final OriginalFile wrongPathDuplicateFile : wrongPathDuplicateFiles) {
+            String fromPathName = wrongPathDuplicateFile.getPath() + wrongPathDuplicateFile.getName();
+            final boolean isLog = fromPathName.endsWith(".log");
```

Comment:
Do log files ever get gzipped? cc: @sbesson

Reply:
Not that I've ever seen; there is also the worry about casting the net too wide in trying to find fileset files outside the fileset's directory. (Would be much easier if people would just store their logfiles, pyramids, uncompressed TIFFs, etc. on a compressing filesystem!) At least this part should be easily tweaked as the occasion demands.